src/Groups/MatGrp.jl

```diff
@@ -0,0 +1,888 @@
+import AbstractAlgebra: MatElem, matrix, MatSpace, parent_type, Ring, RingElem
+import Hecke: base_ring, det, fmpz, fq_nmod, FqNmodFiniteField, nrows, tr, trace
+import GAP: FFE
+
+export
+    general_linear_group,
+    mat_elem_type,
+    matrix_group,
+    MatrixGroup,
+    MatrixGroupElem,
+    omega_group,
+    orthogonal_group,
+    ring_elem_type,
+    special_linear_group,
+    special_orthogonal_group,
+    special_unitary_group,
+    symplectic_group,
+    unitary_group,
+    GL, GO, GU, SL, SO, Sp, SU
+
+
+
+abstract type AbstractMatrixGroupElem <: GAPGroupElem{GAPGroup} end
+
+# NOTE: always defined are deg, ring and at least one between { X, gens, descr }
+# NOTE: the fields ring_iso and mat_iso are always defined if the field X is
+"""
+    MatrixGroup{RE<:RingElem, T<:MatElem{RE}} <: GAPGroup
+Type of groups `G` of `n x n` matrices over the ring `R`, where `n = degree(G)` and `R = base_ring(G)`.
+
+At the moment, only rings of type `FqNmodFiniteField` are supported.
+"""
+mutable struct MatrixGroup{RE<:RingElem, T<:MatElem{RE}} <: GAPGroup
+   deg::Int
+   ring::Ring
+   X::GapObj
+   gens::Vector{<:AbstractMatrixGroupElem}
+   descr::Symbol                       # e.g. GL, SL, symbols for isometry groups
+   ring_iso::GenRingIso
+   mat_iso::GenMatIso
+   order::fmpz
```

Comment:
Where is this used or set?

Reply:
`:descr` is set during the construction, and it is used in the `show` method and in the setting of `:X`.
`:ring_iso` and `:mat_iso` are set when the field `:X` is set, or when the field `:X` of any element of `G` is set.

Reply:
OK, sure, but I was specifically wondering about the line I commented on: what is `order` needed for?

Reply:
It serves just to store the value (similarly to what happens in GAP objects) and avoid expensive re-computations. I introduced it for the cases the order can be computed without using GAP procedures (there are not examples in this branch, but there are, for example, in the branch `jordan-struct`, where the order of centralizers in SL and GL are computed independently).
The alternative would be to set the field `:X` (the underlying GAP object) whenever call the function `order` and using `SetSize` on the GAP object `G.X`.